crates/oxc_ast/src/ast/ts.rs

```diff
@@ -41,10 +41,20 @@ export interface TSIndexSignatureName extends Span {
 pub struct TSThisParameter<'a> {
     #[serde(flatten)]
     pub span: Span,
-    pub this: IdentifierName<'a>,
+    pub this: TSThisParameterKeyword,
     pub type_annotation: Option<Box<'a, TSTypeAnnotation<'a>>>,
 }
 
+#[ast]
+#[derive(Debug, Hash)]
+#[generate_derive(CloneIn, GetSpan, GetSpanMut)]
+#[cfg_attr(feature = "serialize", derive(Serialize, Tsify))]
+#[serde(tag = "type", rename_all = "camelCase")]
+pub struct TSThisParameterKeyword {
```

Comment:
Do we need a dedicated ast node at all? I'd either keep `IdentifierName` or use just a `Span`, otherwise it's just bike shedding at this point ...

Reply:
We could just have `keyword_span: Span` field instead. But that's not a pattern we use anywhere else in the AST, so I thought better to have a node for it.

But if you prefer that, sure.

Reply:
Let's just have a span :-) I want less AST nodes.